src/Compatibility/ApiCompat/Microsoft.DotNet.PackageValidation/Filtering/TargetFrameworkRegexFilter.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.PackageValidation.Filtering
+{
+    /// <summary>
+    /// Helper to check for excluded target frameworks based on a regex pattern.
+    /// </summary>
+    /// <param name="excludeTargetFrameworks">String pattern to exclude target frameworks. Multiple target frameworks must be separated with the ';' character. The glob '*' character is supported.</param>
+    public class TargetFrameworkRegexFilter(string excludeTargetFrameworks) : ITargetFrameworkRegexFilter
+    {
+        private const char TargetFrameworkDelimiter = ';';
+        private readonly Regex? _excludeTargetFrameworks = TransformPatternsToRegexList(excludeTargetFrameworks);
+        private readonly HashSet<string> _foundExcludedTargetFrameworks = [];
+
+        /// <inheritdoc/>
+        public IReadOnlyCollection<string> FoundExcludedTargetFrameworks => _foundExcludedTargetFrameworks;
+
+        /// <inheritdoc/>
+        public bool IsExcluded(string targetFramework)
```

Comment:
Would it make sense to have a `NuGetFramework` overload as well to ensure you're always comparing the same string format rather than relying on the caller to convert to string in the same way?

Reply:
Great idea. Done.